haxe/ui/components/Button.hx

```diff
@@ -200,7 +200,7 @@ class ButtonLayout extends DefaultLayout {
         }
         var size = super.calcAutoSize(exclusions);
         if (itemRenderer != null && isIconRelevant()) {
```

Comment:
* for the extra space in the other side -> this is what will actually make the extra space

        if (itemRenderer == null && isIconRelevant()) {
            if (label != null && label.textAlign == "center" && (iconPosition == "far-right" || iconPosition == "far-left" || iconPosition == "center-right" || iconPosition == "center-left")) {
                size.width += icon.width + horizontalSpacing;
            }
        }